worlds/stardew_valley/logic/goal_logic.py

```diff
@@ -0,0 +1,176 @@
+                continue
+            if location.mod_name and location.mod_name not in self.options.mods:
+                continue
+            all_recipes_names.append(location.name[len(cooksanity_prefix):])
+        all_recipes = [all_cooking_recipes_by_name[recipe_name] for recipe_name in all_recipes_names]
+        return self.logic.and_(*(self.logic.cooking.can_cook(recipe) for recipe in all_recipes))
+
+    def can_complete_craft_master(self) -> StardewRule:
+        craftsanity_prefix = "Craft "
+        all_recipes_names = []
+        exclude_island = self.options.exclude_ginger_island == options.ExcludeGingerIsland.option_true
+        exclude_masteries = not self.content.features.skill_progression.are_masteries_shuffled
+        for location in locations_by_tag[LocationTags.CRAFTSANITY]:
+            if not location.name.startswith(craftsanity_prefix):
+                continue
+            if exclude_island and LocationTags.GINGER_ISLAND in location.tags:
+                continue
+            # FIXME Remove when recipes are in content packs
+            if exclude_masteries and LocationTags.REQUIRES_MASTERIES in location.tags:
+                continue
+            if location.mod_name and location.mod_name not in self.options.mods:
+                continue
+            all_recipes_names.append(location.name[len(craftsanity_prefix):])
+        all_recipes = [all_crafting_recipes_by_name[recipe_name] for recipe_name in all_recipes_names]
+        return self.logic.and_(*(self.logic.crafting.can_craft(recipe) for recipe in all_recipes))
+
+    def can_complete_legend(self) -> StardewRule:
+        return self.logic.money.can_have_earned_total(10_000_000)
+
+    def can_complete_mystery_of_the_star_drop(self) -> StardewRule:
+        other_rules = []
+        number_of_stardrops_to_receive = 0
+        number_of_stardrops_to_receive += 1  # The Mines level 100
+        number_of_stardrops_to_receive += 1  # Old Master Cannoli
+        number_of_stardrops_to_receive += 1  # Museum Stardrop
+        number_of_stardrops_to_receive += 1  # Krobus Stardrop
+
+        # Master Angler Stardrop
+        if self.content.features.fishsanity.is_enabled:
+            number_of_stardrops_to_receive += 1
+        else:
+            other_rules.append(self.logic.fishing.can_catch_every_fish())
+
+        if self.options.festival_locations == options.FestivalLocations.option_disabled:  # Fair Stardrop
+            other_rules.append(self.logic.season.has(Season.fall))
+        else:
+            number_of_stardrops_to_receive += 1
+
+        # Spouse Stardrop
+        if self.content.features.friendsanity.is_enabled:
+            number_of_stardrops_to_receive += 1
+        else:
+            other_rules.append(self.logic.relationship.has_hearts_with_any_bachelor(13))
+
+        if ModNames.deepwoods in self.options.mods:  # Petting the Unicorn
+            number_of_stardrops_to_receive += 1
+
+        if not other_rules:
```

Comment:
`and_()` does not automatically resolve to true, to avoid potential mistakes. But it is true that `&` would have optimized away if one factor is true.

Reply:
Maybe I could add an `allow_empty` optional parameter, to support use cases like this 🤔